File: frontend/src/types/resume.ts
export interface Education {
  id: number
  institution: string
  degree?: string | null
  field_of_study?: string | null
  start_date?: string | null
  end_date?: string | null
  description?: string | null
  position: number
}

export interface Experience {
  id: number
  company: string
  title: string
  location?: string | null
  start_date?: string | null
  end_date?: string | null
  description?: string | null
  position: number
}

export interface Project {
  id: number
  name: string
  description?: string | null
  url?: string | null
  position: number
}

export interface Skill {
  id: number
  name: string
  category?: string | null
  position: number
}

export interface Certification {
  id: number
  name: string
  issuer?: string | null
  date_obtained?: string | null
  url?: string | null
  position: number
}

export interface Resume {
  id: number
  title: string
  full_name?: string | null
  email?: string | null
  phone?: string | null
  location?: string | null
  website?: string | null
  linkedin?: string | null
  github?: string | null
  summary?: string | null
  created_at: string
  updated_at: string
  education: Education[]
  experience: Experience[]
  projects: Project[]
  skills: Skill[]
  certifications: Certification[]
}

export interface ResumeSummary {
  id: number
  title: string
  created_at: string
  updated_at: string
}

export interface ResumeInput {
  title?: string
  full_name?: string
  email?: string
  phone?: string
  location?: string
  website?: string
  linkedin?: string
  github?: string
  summary?: string
}

export interface EducationInput {
  institution: string
  degree?: string
  field_of_study?: string
  start_date?: string
  end_date?: string
  description?: string
}

export interface ExperienceInput {
  company: string
  title: string
  location?: string
  start_date?: string
  end_date?: string
  description?: string
}

export interface ProjectInput {
  name: string
  description?: string
  url?: string
}

export interface SkillInput {
  name: string
  category?: string
}

export interface CertificationInput {
  name: string
  issuer?: string
  date_obtained?: string
  url?: string
}

File: backend/app/models/resume.py
from sqlalchemy import Column, Integer, String, Text, DateTime, ForeignKey, func
from sqlalchemy.orm import relationship

from app.database.base import Base


class Resume(Base):
    __tablename__ = "resumes"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id", ondelete="CASCADE"), nullable=False, index=True)
    title = Column(String(200), nullable=False, default="Untitled Resume")
    full_name = Column(String(100), nullable=True)
    email = Column(String(254), nullable=True)
    phone = Column(String(30), nullable=True)
    location = Column(String(100), nullable=True)
    website = Column(String(2048), nullable=True)
    linkedin = Column(String(2048), nullable=True)
    github = Column(String(2048), nullable=True)
    summary = Column(Text, nullable=True)
    created_at = Column(DateTime, nullable=False, server_default=func.now())
    updated_at = Column(DateTime, nullable=False, server_default=func.now(), onupdate=func.now())

    user = relationship("User", back_populates="resumes")
    education = relationship("EducationEntry", back_populates="resume", cascade="all, delete-orphan", order_by="EducationEntry.position")
    experience = relationship("ExperienceEntry", back_populates="resume", cascade="all, delete-orphan", order_by="ExperienceEntry.position")
    projects = relationship("ProjectEntry", back_populates="resume", cascade="all, delete-orphan", order_by="ProjectEntry.position")
    skills = relationship("Skill", back_populates="resume", cascade="all, delete-orphan", order_by="Skill.position")
    certifications = relationship("Certification", back_populates="resume", cascade="all, delete-orphan", order_by="Certification.position")
    applications = relationship("JobApplication", back_populates="resume")
